### PR TITLE
絞り込み機能の実装

### PR DESCRIPTION
**Pull request by yamada-kensyu**
レビューよろしくお願いします。

### CODE REVIEW

File: src/main/java/jp/co/example/ecommerce_a/repository/ItemRepository.java [hunk 1]
```diff
@@ -71,15 +71,15 @@ public List<Item> findByTargetAgeGenreAndPriceRange(Integer targetAge, Integer g
 		MapSqlParameterSource addedParam = new MapSqlParameterSource();
 		if (targetAge!=null) {
 			sqlForTargetAge = " target_age=:targetAge";
-			addedParam.addValue("target_age", targetAge);
+			addedParam.addValue("targetAge", targetAge);
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/repository/ItemRepository.java:74]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' child has incorrect indentation level 24, expected level should be 6.

File: src/main/java/jp/co/example/ecommerce_a/repository/ItemRepository.java [hunk 2]
```diff
@@ -71,15 +71,15 @@ public List<Item> findByTargetAgeGenreAndPriceRange(Integer targetAge, Integer g
...
 		}
 		if (genre!=null) {
 			sqlForGenre = " AND genre=:genre";
 			addedParam.addValue("genre", genre);
 		}
 		if (priceRange!=null) {
 			sqlForPriceRange = " AND price_range=:priceRange;";
-			addedParam.addValue("price_range", priceRange);
+			addedParam.addValue("priceRange", priceRange);
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/repository/ItemRepository.java:82]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' child has incorrect indentation level 24, expected level should be 6.

File: src/main/java/jp/co/example/ecommerce_a/common/SecurityConfig.java [hunk 3]
```diff
@@ -25,7 +25,7 @@ SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
 		// アクセスの制限に関する設定
 		http.authorizeHttpRequests(ahr -> ahr
 				.requestMatchers("/", "/registerUser/**", "/login/**", "/showItemDetail/**",
-						"/showShoppingCart/**", "/addShoppingCart/**", "/deleteShoppingCart/**")
+						"/showShoppingCart/**", "/addShoppingCart/**", "/deleteShoppingCart/**","/narrowingDown")
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/common/SecurityConfig.java:28]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/common/SecurityConfig.java:28]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>',' is not followed by whitespace.

**Review by quangtung98 — APPROVED** [review]
問題がなさそうと思いますので、認証します。ありがとうございました。

**Pull request merged**